src/test/kotlin/uk/gov/dluhc/printapi/messaging/ProcessPrintResponseMessageListenerIntegrationTest.kt

```diff
@@ -138,7 +139,80 @@ internal class ProcessPrintResponseMessageListenerIntegrationTest : IntegrationT
+            wireMockService.verifyEroManagementGetEro(expectedGssCode)
+            assertEmailSent(expectedEmailMessage)
+        }
+    }
+
+    @Test
+    fun `should process print response message and send Certificate Failed To Print email`() {
+        // Given
+        val requestId = aValidRequestId()
+        val batchId = aValidBatchId()
+        val certificate = buildCertificate(
+            status = Status.ASSIGNED_TO_BATCH,
+            printRequests = mutableListOf(
+                buildPrintRequest(
+                    batchId = batchId,
+                    requestId = requestId,
+                    printRequestStatuses = listOf(
+                        buildPrintRequestStatus(
+                            status = Status.ASSIGNED_TO_BATCH,
+                            eventDateTime = Instant.now().minusSeconds(10)
+                        )
+                    )
+                )
+            )
+        )
+        certificateRepository.save(certificate)
+
+        val message = buildProcessPrintResponseMessage(
+            requestId = requestId,
+            status = ProcessPrintResponseMessage.Status.FAILED,
+            statusStep = ProcessPrintResponseMessage.StatusStep.IN_MINUS_PRODUCTION,
+        )
+        val expectedStatus = Status.PRINT_PROVIDER_PRODUCTION_FAILED
+
+        val expectedGssCode = certificate.gssCode!!
+        val ero = buildElectoralRegistrationOfficeResponse(
+            localAuthorities = listOf(
+                buildLocalAuthorityResponse(
+                    gssCode = expectedGssCode,
+                    contactDetailsEnglish = buildContactDetails(emailAddress = "a-user@valtech.com")
+                ),
+            )
+        )
+        wireMockService.stubEroManagementGetEroByGssCode(ero, expectedGssCode)
+
+        val expectedEmailMessage = buildLocalstackEmailMessage(
+            timestamp = OffsetDateTime.now(UTC).toLocalDateTime().truncatedTo(SECONDS),
+            emailSender = "noreply_erouser@erop.ierds.uk",
+            toAddresses = setOf("a-user@valtech.com"),
+            subject = "Electoral Registration Office Portal - printing failed - application ${certificate.applicationReference}",
+            htmlBody = """
+                <html>
+                  <body>
+                    <p>Printing has failed for the application ${certificate.applicationReference}</p>
+                    <p>You can access the application here:</p>
+                    <p><a href="https://erop.ierds.uk/voter-authority-certificate/${certificate.applicationReference}">
+                                https://erop.ierds.uk/voter-authority-certificate/${certificate.applicationReference}</a>
+                    </p>
+                    <br>
+                    <p>This is an automated message from the ERO Portal.  You will not be able to reply to this email.</p>
+                  </body>
+                </html>
+            """.lines().joinToString(separator = "\\s*") { it.trim() }
```

Comment:
Minor - this is very thorough, but also very brittle (perhaps deliberately?). At alternative would be to check for the dynamic values only.

Reply:
I've simplified this to just test for the application reference.